fem/fe.cpp

```diff
@@ -1057,7 +1078,31 @@ void VectorFiniteElement::Project_RT(
    }
    else
    {
-      mfem_error("VectorFiniteElement::Project_RT (fe version)");
+      int sdim = Trans.GetSpaceDim();
+      double vk[Geometry::MaxDim];
+      DenseMatrix vshape(fe.GetDof(), sdim);
+      Vector vshapenk(fe.GetDof());
+      const bool square_J = (dim == sdim);
+
+      I.SetSize(dof, fe.GetDof());
+      for (int k = 0; k < dof; k++)
+      {
+         const IntegrationPoint &ip = Nodes.IntPoint(k);
+
+         Trans.SetIntPoint(&ip);
+         // Transform RT face normals from reference to physical space
+         // vk = adj(J)^T nk
+         Trans.AdjugateJacobian().MultTranspose(nk + d2n[k]*dim, vk);
+         // Compute fe basis functions in physical space
+         fe.CalcVShape(Trans, vshape);
+         // Project fe basis functions onto transformed face normals
+         vshape.Mult(vk, vshapenk);
+         if (!square_J) { vshapenk /= Trans.Weight(); }
```

Comment:
Has this been tested in the non-square case, e.g. 2D mesh in 3D space? It would be helpful to document this weight (what is the transformation and why is the weight needed only in the square case; and note that the `Weight()` is `det` in the square case, something else in the non-square case).

Reply:
This is a good question.  I'm not aware of any places where we test the support of curves in 2D or 3D and surfaces in 3D.  We should certainly add unit tests for these cases but that's beyond the scope of this PR.

I'm not sure of the best place to document this (nor am I certain I fully understand it).  We should probably document `DenseMatrix::CalcAdjugate`, `DenseMatrix::Weight`, and `ElementTransformation::AdjugateJacobian` but people would still have questions.

I'm tempted to kick this can down the road if you don't mind.

Reply:
Ok, we should document this in the future, and I agree the documentation needs to be in several places beyond this function. I have used 2D spaces in 3D in my DD code (on interfaces), so I figured this out at one point (but don't recall all the details now). Maybe I will review it soon and try to document everything.